### PR TITLE
Add latency metrics to SDK

### PR DESCRIPTION
**Pull request by rtso**
## Description

- Add transaction latency metrics for AsyncStep  and PollableAsyncStep. It is calculated using (current time - txn timestamp) 
- This will be used in Forge tests to track total processor latency

## Testing
Tested with example processor

### CODE REVIEW

**Comment by rtso** [comment]
* **#67** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-indexer-processor-sdk/67?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rtso and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: aptos-indexer-processors-sdk/sdk/src/utils/step_metrics.rs [hunk 1]
```diff
@@ -23,6 +23,12 @@ pub fn init_step_metrics_registry(registry: &mut Registry) {
         LATEST_PROCESSED_TRANSACTION_TIMESTAMP.clone(),
     );
 
+    registry.register(
+        format!("{}_{}", METRICS_PREFIX, "processed_transaction_latency"),
```
Comment (dermanyang) [inline aptos-indexer-processors-sdk/sdk/src/utils/step_metrics.rs:27]: should we include units here?

**Review by dermanyang — APPROVED** [review]

File: aptos-indexer-processors-sdk/sdk/src/utils/step_metrics.rs [hunk 2]
```diff
@@ -23,6 +23,12 @@ pub fn init_step_metrics_registry(registry: &mut Registry) {
...
+        "Latency of the polled transactions, computed by (txn timestamp - current time)",
```
Comment (rustielin) [inline aptos-indexer-processors-sdk/sdk/src/utils/step_metrics.rs:28]: does this description need to be updated?

**Review by rustielin — APPROVED** [review]
Stamping to unblock, but take a look at the comments above ^

**Pull request merged**